packages/framework/ai-collab/src/explicit-strategy/typeGeneration.ts

```diff
@@ -362,13 +435,34 @@ function getTypeForAllowedTypes(
 		);
 	}
 }
+
 function tryGetSingleton<T>(set: ReadonlySet<T>): T | undefined {
 	if (set.size === 1) {
 		for (const item of set) {
 			return item;
 		}
 	}
 }
+
 function hasAtLeastTwo<T>(array: T[]): array is [T, T, ...T[]] {
 	return array.length >= 2;
 }
+
+/**
+ * Determines if the provided {@link TreeNode} contains an array schema.
+ */
+export function doesNodeContainArraySchema(node: TreeNode): boolean {
+	const schema = Tree.schema(node);
+	const simpleSchema = getSimpleSchema(schema);
+
+	const definitionMap = simpleSchema.definitions;
+
+	for (const name of simpleSchema.definitions.keys()) {
+		const nodeSchema = definitionMap.get(name) ?? fail("Unexpected definition");
+		if (nodeSchema.kind === NodeKind.Array) {
+			return true;
+		}
+	}
```

Comment:
I think this works? The "Unexpected definition" check seems unnecessary since we're just iterating over the keys we know exist and there's no async-ness here that could allow another bit of code to mutate `simpleSchema.definitions` before this function returns.

```suggestion
	for (const [name, nodeSchema] of simpleSchema.definitions) {
		if (nodeSchema.kind === NodeKind.Array) {
			return true;
		}
	}
```

Reply:
Yes that pattern was something left over that I brought in and I agree it's unnecessary for this function. Updated in commit c595f44eb33065af14eb1b92eccbf7eb4b35ea09